aw-transform/src/flood.rs

```diff
@@ -36,25 +38,36 @@ pub fn flood(events: Vec<Event>, pulsetime: chrono::Duration) -> Vec<Event> {
                     warn!("Gap was of negative duration ({}s), but could be safely merged. This error will only show once per batch.", gap);
                     warned_negative_gap_safe = true;
                 }
-                // Extend e1 to the middle between e1 and e2
-                e1.duration = e2.calculate_endtime() - e1.timestamp;
+                // Choose the longest event and set the endtime to it
+                let e1_endtime = e1.calculate_endtime();
+                let e2_endtime = e2.calculate_endtime();
+                if e2_endtime > e1_endtime {
+                    e1.duration = e2_endtime - e1.timestamp;
+                } else {
+                    e1.duration = e1_endtime - e1.timestamp;
+                }
                 // Drop next event since they are merged and flooded into e1
-                drop_next = true;
+                e1_iter.next();
+                // Retry this event again to give it a change to merge e1
+                // with 'e3'
+                retry_e = Some(e1);
+                // Since we are retrying on this event we don't want to push it
+                // to the new_events vec
+                continue;
```

Comment:
Cool! Took me a while to understand what's going on with `retry_e`, but looks performant :smiley:

Reply:
It's mostly to make the borrow checker happy, there are more readable ways to implement it if you write "unsafe" code.